### DISCUSSION TITLE
Filter error in maps

### DISCUSSION THREAD

**Discussion by workualemu** [tech-acs/chimera-docs#11] [Bug report] [answered]
The filter for maps returns a value of different format than that of the charts.  For instance, in Namibia, the filter for charts at constituency level for Region '12' and Constituency '1207' looks like the following:
```
   Region: '12'
   Constituency: '12.1207'
```

But the same filter for maps is:
```
   Region: '12'
   Constituency: '1207'
```

Because of this issue, areas cannot be returned as expected

**Comment by amestsantim** [comment, marked as answer]
Release 2.7.4 (https://github.com/tech-acs/chimera-starter-kit/releases/tag/2.7.4) makes the Map API more like the Chart API
In the MapIndicator implementation class, implement getData just like in Chart and use the 
getMappableData(Collection $data, string $filterPath): Collection 
as the equivalent of getTraces. 
It received the data collection returned from getData and then you can mutate that collection and merge in reference values, which you can inturn pull in using the passed in $filterPath parameter
If you do override it, make sure you either implement the data "shaping" that the base method does, or just call the base class method by passing in your mutated collection.
